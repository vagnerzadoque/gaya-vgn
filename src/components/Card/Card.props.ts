export interface CardProps {

  /**
   * The content of Card
   */
  children: React.ReactNode

  /**
   * Override or extend the styles applied to the component
   */
  className?: string

  /**
   * If `false`, a box-shadow effect are disabled
   * @default true
   */
  elevation?: boolean;

  /**
   * If `false`, rounded corners are disabled
   * @default true
   */
  radius?: boolean;

  /**
   * Optional ID for testing
   */
  testID?: string;
}
