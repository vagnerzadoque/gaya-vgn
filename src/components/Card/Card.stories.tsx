import React from 'react'
import type { Meta, StoryObj } from '@storybook/react'
import { fn } from '@storybook/test'
import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { Card, CardProps } from '.'
import { Button } from '../Button'
import { Icon } from '../Icon'
import StoryContainer from '../../helpers/StoryContainer'

const componentStatus = `
> Cards are surfaces that groups content and actions in the same space. Its construction must always create a visual hierarchy of the content and communicate its purposes

## Properties
| Property                     | Values            |    Status           |
|---                           |                ---|                  ---|
| **elevation**                | true/false        | ✅ Available        |
| **radius**                   | true/false        | ✅ Available        |

## Technical Usages Examples
`

const meta: Meta<CardProps> = {
  title: 'Components/Card',
  component: Card,
  parameters: {
    componentSubtitle: '',
    docs: { description: { component: componentStatus } }
  }
};

export default meta;
type Story = StoryObj<CardProps>;

export const Playground: Story = {
  render: (args) => {
    const { image, wrapper } = containerStyles()
    const { text } = textStyle()

    return (
      <Card {...args}>
        <div className={image}>
          <Icon name="outlined-product-brandsproduct" size="hugeXXX" />
        </div>
        <div className={wrapper}>
          <p className={text}>
            This is just an example of the Card component
            containing an image, a text and a button
          </p>
          <Button onClick={fn()} fullWidth>button</Button>
        </div>
      </Card>
    )
  }
}

export const Elevation: Story = {
  render: (args) => {
    const { text } = textStyle()

    return (
      <StoryContainer>
        <Card {...args}>
          <p className={text}>
            This is an example of the Card component with elevation
          </p>
        </Card>
        <Card {...args} elevation={false}>
          <p className={text}>
            This is an example of the Card component without elevation
          </p>
        </Card>
      </StoryContainer>
    )
  }
}

export const Radius: Story = {
  render: (args) => {
    const { text } = textStyle()

    return (
      <StoryContainer>
        <Card {...args}>
          <p className={text}>
            This is an example of the Card component with radius
          </p>
        </Card>
        <Card {...args} radius={false}>
          <p className={text}>
            This is an example of the Card component without radius
          </p>
        </Card>
      </StoryContainer>
    )
  }
}

const textStyle = createUseStyles((theme: Theme) => ({
  text: {
    color: theme.color.highEmphasis,
    fontFamily: [theme.typography.fontFamily.primary, theme.typography.fontFamily.secondary],
    fontSize: 14,
    fontWeight: 'normal',
    letterSpacing: 0.22,
    maxWidth: 200,
    textAlign: 'center'
  }
}))

const containerStyles = createUseStyles((theme: Theme) => ({
  image: {
    backgroundColor: theme.color.lowEmphasis,
    opacity: 0.4,
    textAlign: 'center'
  },
  wrapper: {
    padding: theme.spacing.small
  }
}))
