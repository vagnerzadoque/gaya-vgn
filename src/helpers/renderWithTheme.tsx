import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import themes from '@naturacosmeticos/natds-themes'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component: JSX.Element) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider theme={themes.natura.light}>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
