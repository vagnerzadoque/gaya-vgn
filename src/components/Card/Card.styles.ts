import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { CardProps } from './Card.props'

type CardStyleProps = Required<Pick<CardProps, 'elevation' | 'radius'>>

const styles = createUseStyles((theme: Theme) => ({
  card: {
    backgroundColor: theme.color.surface,
    borderRadius: ({ radius }: CardStyleProps) => (radius ? theme.borderRadius.medium : 'none'),
    boxShadow: ({ elevation }: CardStyleProps) => (elevation ? theme.elevation.micro : 'none'),
    display: 'inline-block',
    overflow: 'hidden'
  }
}))

export default styles
