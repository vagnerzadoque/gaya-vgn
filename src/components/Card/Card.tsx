import React from 'react'
import { CardProps } from './Card.props'
import styles from './Card.styles'

const Card = React.forwardRef<HTMLDivElement, CardProps>(
  ({
    children,
    elevation = true,
    radius = true,
    testID,
    className = '',
    ...props
  }, ref) => {
    const { card } = styles({ elevation, radius })

    return (
      <div
        className={`${className} ${card} Card-gaya`}
        data-testid={testID}
        ref={ref}
        {...props}
      >
        {children}
      </div>
    )
  }
)

export default Card
