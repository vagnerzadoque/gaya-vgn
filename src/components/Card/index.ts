import Card from './Card'

export { Card }
export { CardProps } from './Card.props'
export default Card
