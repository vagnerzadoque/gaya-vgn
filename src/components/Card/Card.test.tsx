import React from 'react'
import renderWithTheme from '../../helpers/renderWithTheme'
import Card from './Card'

describe('Card component', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<Card>some content</Card>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should not apply box-shadow style', () => {
    const { styles, component } = renderWithTheme(<Card elevation={false}>some content</Card>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })

  it('should not apply border-radius style', () => {
    const { styles, component } = renderWithTheme(<Card radius={false}>some content</Card>)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
