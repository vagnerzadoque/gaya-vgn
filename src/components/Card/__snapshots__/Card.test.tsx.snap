// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should not apply border-radius style 1`] = `
[
  ".card-0-2-1 {
  display: inline-block;
  overflow: hidden;
  background-color: #ffffff;
}
.card-d0-0-2-2 {
  border-radius: none;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.14);
}",
  <div>
    <div
      class=" card-0-2-1 card-d0-0-2-2 Card-gaya"
    >
      some content
    </div>
  </div>,
]
`;

exports[`Card component should not apply box-shadow style 1`] = `
[
  ".card-0-2-1 {
  display: inline-block;
  overflow: hidden;
  background-color: #ffffff;
}
.card-d0-0-2-2 {
  border-radius: 4px;
  box-shadow: none;
}",
  <div>
    <div
      class=" card-0-2-1 card-d0-0-2-2 Card-gaya"
    >
      some content
    </div>
  </div>,
]
`;

exports[`Card component should render correctly with default props 1`] = `
[
  ".card-0-2-1 {
  display: inline-block;
  overflow: hidden;
  background-color: #ffffff;
}
.card-d0-0-2-2 {
  border-radius: 4px;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.14);
}",
  <div>
    <div
      class=" card-0-2-1 card-d0-0-2-2 Card-gaya"
    >
      some content
    </div>
  </div>,
]
`;
